MonkeyWrench.Database/Extensions/DBRevisionWork_Extensions.cs

```diff
@@ -322,35 +322,24 @@ public static DBRevisionWork Find (DB db, DBLane lane, DBHost host, DBRevision r
 		/// <param name="host"></param>
 		public static bool SetWorkHost (this DBRevisionWork rw, DB db, DBHost host)
 		{
-			object result;
-			string update_cmd = string.Format (@"UPDATE RevisionWork SET workhost_id = {0}, assignedtime = NOW() WHERE id = {1} AND workhost_id IS NULL;", host.id, rw.id);
-
-			using (IDbCommand cmd = db.CreateCommand ()) {
-				cmd.CommandText = update_cmd;
-
-				var rv = cmd.ExecuteNonQuery ();
-
-				if (rv != 1) {
-					log.DebugFormat ("{0}: {1} (failed)", cmd.CommandText, rv);
-					return false;
-				}
-			}
-
 			using (IDbCommand cmd = db.CreateCommand ()) {
 				cmd.CommandText = @"
-SELECT workhost_id FROM RevisionWork where id = @id AND workhost_id = @workhost_id;
-";
+					UPDATE RevisionWork
```

Comment:
It wouldn't function if `workhost_id` was assigned to the bot requesting work, and there's no reason for it to do two queries.